### ISSUE TITLE
State update to a falsy value is ignored in isolated component

### ISSUE THREAD

**Issue by abaco** [staltz/cycle-onionify#10]
It seems legal to have as state a value that is not an object, for example an integer, a string or a boolean. Hovever, when such a value is falsy it gets filtered out (https://github.com/staltz/cycle-onionify/blob/master/src/index.ts#L79).

Non-object state values should be either forbidden or fully supported (including falsy values).

**Code to reproduce the issue:**
```js
import xs from 'xstream';
import Cycle from '@cycle/xstream-run';
import {div, button, p, makeDOMDriver} from '@cycle/dom';
import onionify from '../../../lib/index';
import isolate from '@cycle/isolate'

function Counter(sources) {
  const action$ = xs.merge(
    sources.DOM.select('.decrement').events('click').map(ev => -1),
    sources.DOM.select('.increment').events('click').map(ev => +1)
  );

  const state$ = sources.onion.state$;

  const vdom$ = state$.map(state =>
    div([
      button('.decrement', 'Decrement'),
      button('.increment', 'Increment'),
      p('Counter: ' + state)
    ])
  );

  const updateReducer$ = action$.map(num => function updateReducer(prevState) {
    return prevState + num;
  });

  return {
    DOM: vdom$,
    onion: updateReducer$,
  };
}

function main(sources) {
  const counterSinks = isolate(Counter, 'count')(sources);

  const vdom$ = counterSinks.DOM;

  const initReducer$ = xs.of(function initReducer() {
    return {count: 1};
  });
  const updateReducer$ = counterSinks.onion;
  const reducer$ = xs.merge(initReducer$, updateReducer$);

  return {
    DOM: vdom$,
    onion: reducer$,
  };
}

const wrappedMain = onionify(main);

Cycle.run(wrappedMain, {
  DOM: makeDOMDriver('#main-container')
});
```

**Expected behavior:**
Clicking "Decrement" once should change the counter value to 0.

**Actual behavior:**
The counter value is in internally updated, but the isolated component won't see the update. Thus the counter can be set to any value except 0.

**Versions of packages used:**
cycle-onionify: 2.3.0
@cycle/base: 4.1
@cycle/isolate: 1.4

**Note that the following (without isolation) works fine:**
```js
import xs from 'xstream';
import Cycle from '@cycle/xstream-run';
import {div, button, p, makeDOMDriver} from '@cycle/dom';
import onionify from '../../../lib/index';

function main(sources) {
  const action$ = xs.merge(
    sources.DOM.select('.decrement').events('click').map(ev => -1),
    sources.DOM.select('.increment').events('click').map(ev => +1)
  );

  const state$ = sources.onion.state$;

  const vdom$ = state$.map(state =>
    div([
      button('.decrement', 'Decrement'),
      button('.increment', 'Increment'),
      p('Counter: ' + state)
    ])
  );

  const initReducer$ = xs.of(function initReducer() {
    return 0;
  });
  const updateReducer$ = action$.map(num => function updateReducer(prevState) {
    return prevState + num;
  });
  const reducer$ = xs.merge(initReducer$, updateReducer$);

  return {
    DOM: vdom$,
    onion: reducer$,
  };
}

const wrappedMain = onionify(main);

Cycle.run(wrappedMain, {
  DOM: makeDOMDriver('#main-container')
});
```

**Comment by staltz** [comment]
Seems like an obvious bug. We should support falsy values, and do a `typeof x === 'undefined'` check instead of `!!x`.

**Comment by staltz** [comment]
Released v2.4 with this fixed
